pkg/naming/metrics_query_test.go

```diff
@@ -76,7 +76,15 @@ func checks(cs ...checkFunc) checkFunc {
 
 func TestBuildSelector(t *testing.T) {
 	mustNewQuery := func(queryTemplate string, namespaced bool) MetricsQuery {
-		mq, err := NewMetricsQuery(queryTemplate, &resourceConverterMock{namespaced})
+		mq, err := NewMetricsQuery(queryTemplate, &resourceConverterMock{namespaced}, true)
```

Comment:
same as above, we only want to check the `namespaced` value for BuildExtrernal as this feature is only meant for external metrics.

Reply:
Agreed. But both both forms of metrics use the same type which is why all of the calls the `NewMetricsQuery` have to take a namespace.

We could do something else though:
1. Split `MetricsQuery` into two types: `MetricsQuery` & `ExternalMetricsQuery` . Each with it's own `NewX` function. This would be the cleanest but will result in more code changes
2. Add setter func for `namespaced` so we can change it on after calling `NewMetricsQuery`. This would work but doesn't feel like idomatic go
3. Leave the bare bool in `NewMetricsQuery` as-is

I'm fine with whatever you choose.

Reply:
Splitting into two separate constructors sounds like the best option here.